addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js

```diff
@@ -125,6 +126,123 @@ describe('ProviderService', () => {
     });
   });
 
+  describe('updateUser', () => {
+    it('should update existing user with missing attribute mappings', async () => {
+      // BUILD
+      userService.updateUser = jest.fn();
+      const systemContext = getSystemRequestContext();
+      const username = 'test@example.com';
+      const userAttributes = {
+        username,
+        usernameInIdp: username,
+        identityProviderName: 'Cognito Native Pool',
+        isSamlAuthenticatedUser: false,
+        isNativePoolUser: true,
+        firstName: 'John',
+        lastName: 'Doe',
+        email: 'test@example.com',
+      };
+      const existingUser = {
+        identityProviderName: 'Cognito Native Pool',
+        uid: 'sample-user-uid',
+        username,
+        isNativePoolUser: true,
+        usernameInIdp: username,
+        email: username,
+        rev: 2,
+      };
+      const expected = {
+        firstName: 'John',
+        lastName: 'Doe',
+        isSamlAuthenticatedUser: false,
+        rev: 2,
+        uid: 'sample-user-uid',
+      };
+
+      // OPERATE
+      await service.updateUser(userAttributes, existingUser);
+
+      // CHECK
+      expect(userService.updateUser).toHaveBeenCalled();
+      expect(userService.updateUser).toHaveBeenCalledWith(systemContext, expected);
+    });
+
+    it('should update existing user with updated names in attribute mappings', async () => {
+      // BUILD
+      userService.updateUser = jest.fn();
+      const systemContext = getSystemRequestContext();
+      const username = 'test@example.com';
+      const userAttributes = {
+        username,
+        usernameInIdp: username,
+        identityProviderName: 'Cognito Native Pool',
+        isSamlAuthenticatedUser: false,
+        isNativePoolUser: true,
+        firstName: 'Jen@example.com',
+        lastName: 'Doe',
+        email: 'test@example.com',
+      };
+      const existingUser = {
+        identityProviderName: 'Cognito Native Pool',
+        uid: 'sample-user-uid',
+        username,
+        firstName: 'newUser',
+        lastName: 'Ms. Doe',
+        isNativePoolUser: true,
+        isSamlAuthenticatedUser: false,
+        usernameInIdp: username,
+        email: username,
+        rev: 2,
+      };
+      const expected = {
+        firstName: 'Jen',
+        lastName: 'Doe',
+        rev: 2,
+        uid: 'sample-user-uid',
+      };
```

Comment:
Can you add a test case that matches the setup that led to this bug being found? The values would be something like this:
```
// from cognito
const userAttributes = {
        username,
        usernameInIdp: username,
        identityProviderName: 'Cognito Native Pool',
        isSamlAuthenticatedUser: false,
        isNativePoolUser: true,
        firstName: 'Jen',
        lastName: 'Doe',
        email: 'jen+doe@example.com',
      };
// From DDB
const existingUser = {
        identityProviderName: 'Cognito Native Pool',
        uid: 'sample-user-uid',
        username,
        firstName: 'jen+doe',
        lastName: 'jen+doe',
        isNativePoolUser: true,
        isSamlAuthenticatedUser: false,
        usernameInIdp: username,
        email: username,
        rev: 2,
};
const expected = {
        firstName: 'Jen',
        lastName: 'Doe',
        rev: 2,
        uid: 'sample-user-uid',
};
```

Reply:
This test case is exactly the same (basically we update the names if they're not equal to IdP-provided values).

Reply:
This test case also covers the case where name value sent by IdP is in email format.

Reply:
Added test case